src/OrleansProviders/Streams/Common/PooledCache/PooledQueueCache.cs

```diff
@@ -0,0 +1,306 @@
+﻿
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using Orleans.Streams;
+
+namespace Orleans.Providers.Streams.Common
+{
+    public class PooledQueueCache<TQueueMessage, TCachedMessage>
```

Comment:
Description of what this class does? Due to the double layer of pooling and also different objects for metadata and the message payload, it is a little confusing to follow (at least in the diff). Would it be possible to have a place somewhere that puts all the pieces together? Maybe in ObjectPool which I assume should be the "root" of all this hierarchy? Or if there's a better place, go for it